src/hotspot/os/linux/cgroupSubsystem_linux.hpp

```diff
@@ -154,63 +183,44 @@ template <typename T> int subsystem_file_line_contents(CgroupController* c,
     return 0;
   }
   log_debug(os, container)("Type %s (key == %s) not found in file %s", scan_fmt,
-                           (key == nullptr ? "null" : key), absolute_path);
+                           key, absolute_path);
   return OSCONTAINER_ERROR;
 }
 PRAGMA_DIAG_POP
 
-// log_fmt can be different than scan_fmt. For example
-// cpu_period() for cgv2 uses log_fmt='%d' and scan_fmt='%*s %d'
-#define GET_CONTAINER_INFO(return_type, subsystem, filename,              \
-                           logstring, log_fmt, scan_fmt, variable)        \
-  return_type variable;                                                   \
-{                                                                         \
-  int err;                                                                \
-  err = subsystem_file_line_contents(subsystem,                           \
-                                     filename,                            \
-                                     nullptr,                             \
-                                     scan_fmt,                            \
-                                     &variable);                          \
-  if (err != 0) {                                                         \
-    log_trace(os, container)(logstring "%d", OSCONTAINER_ERROR);          \
-    return (return_type) OSCONTAINER_ERROR;                               \
-  }                                                                       \
-                                                                          \
-  log_trace(os, container)(logstring log_fmt, variable);                  \
-}
+PRAGMA_DIAG_PUSH
+PRAGMA_FORMAT_NONLITERAL_IGNORED
+template <typename T> int cg_file_multi_line_ctrl(CgroupController* c,
```

Comment:
This always bothered me: If the first argument is a pointer to a `CgroupController`, then why isn't this just a method for that object? No need to for the `c == nullptr` check then either.

Reply:
Good question. I hadn't considered it. Let me explore that option a bit and I'll get back to you.